File: day6/code.cpp
#include <bits/stdc++.h>
using namespace std;

void task1(vector<int> t, vector<int> d) {
    long long ans = 1;
    for (int i = 0; i < t.size(); ++i) {
        int cnt = 0;
        for (int k = 0; k <= t[i]; ++k)
            cnt += ((t[i] - k) * k > d[i]);
        ans *= cnt;
    }
    cout << "task1: " << ans << endl;
}

void task2(vector<int> tt, vector<int> dd) {
    string ts, ds;
    for (auto x : tt)
        ts += to_string(x);
    for (auto x : dd)
        ds += to_string(x);
    long long t = stoll(ts), d = stoll(ds);
    long long cnt = 0;
    for (long long k = 0; k <= t; ++k)
        cnt += ((t - k) * k > d);
    cout << "task2: " << cnt << endl;
}

int main() {
    freopen("input.txt", "r", stdin);
    string ts, ds;
    getline(cin, ts);
    getline(cin, ds);
    ts = ts.substr(ts.find_first_of("01234565789"));
    ds = ds.substr(ds.find_first_of("01234565789"));
    stringstream st(ts), sd(ds);
    vector<int> t, d;
    while (st.good())
        t.emplace_back(), st >> t.back();
    while (sd.good())
        d.emplace_back(), sd >> d.back();
    
    task1(t, d);
    task2(t, d);
}

File: day6/code.py
import re

def task1():
    f = open("input.txt")
    ts = f.readline()
    t = []
    while (ts != ''):
        spos = re.search("\d", ts).start()
        l = len(ts[spos:].split(' ')[0])
        t += [int(ts[spos:spos + l])]
        ts = ts[spos + l:]

    ds = f.readline()
    d = []
    while (ds != ''):
        spos = re.search("\d", ds).start()
        l = len(ds[spos:].split(' ')[0])
        d += [int(ds[spos:spos + l])]
        ds = ds[spos + l:]

    ans = 1
    for i in range(len(d)):
        cnt = 0
        for k in range(t[i] + 1):
            cnt += ((t[i] - k) * k > d[i])
        ans *= cnt
    print("task1:", ans)
    
def task2():
    f = open("input.txt")
    ts = f.readline()
    tss = ''
    while (ts != ''):
        spos = re.search("\d", ts).start()
        l = len(ts[spos:].split(' ')[0])
        tss += ts[spos:spos + l]
        ts = ts[spos + l:]
    t = int(tss)

    ds = f.readline()
    dss = ''
    while (ds != ''):
        spos = re.search("\d", ds).start()
        l = len(ds[spos:].split(' ')[0])
        dss += ds[spos:spos + l]
        ds = ds[spos + l:]
    d = int(dss)

    cnt = 0
    for k in range(t + 1):
        cnt += ((t - k) * k > d)
    print("task1:", cnt)

task1()
task2()